### DISCUSSION TITLE
feedParameters doesn't work?

### DISCUSSION THREAD

**Discussion by jeff-h** [sparkle-project/Sparkle#2329] [Q&A]
I'm trying out the [`-feedParametersForUpdater:sendingSystemProfile:`](https://sparkle-project.org/documentation/api-reference/Protocols/SPUUpdaterDelegate.html#/c:objc(pl)SPUUpdaterDelegate(im)feedParametersForUpdater:sendingSystemProfile:) method, but it appears to only add `null`s to the request query parameters.

Running Sparkle 2.3.0.

My `feedParameters` implementation:

```swift
func feedParameters(for updater: SPUUpdater, sendingSystemProfile sendingProfile: Bool) -> [[String : String]] {
    return [
        ["one": "1"],
        ["two": "2"]
    ]
}
```

I've confirmed the method is being called as expected, but the resulting request ends up like this:

`https://myserver.com/stable/appcast.xml?(null)=(null)&(null)=(null)`

Has anyone else been able to get this to work?

**Comment by zorgiepoo** [comment]
> An array of dictionaries with keys: key, value, displayKey, displayValue, the latter two being specifically for display to the user.

Try something like:

```
return [["key": "one", "value": "1", "displayKey": "keyDisplayedToUser", "displayValue": "valueDisplayedToUser"]]
```

They can be displayed in the permission prompt for checking for updates, but maybe only if sending system profile information is enabled. Anyway, this is the API.

**Reply by jeff-h** [reply]
Ohh, I see now! I re-checked the docs and see the expected dictionary keys all described there as you said above.

Thanks, I really appreciate your time responding.
